### PR TITLE
Add methods QPDFTokenizer::Token::isInteger and isWord

### PR DESCRIPTION
**Pull request by m-holger**

### CODE REVIEW

File: include/qpdf/QPDFTokenizer.hh [hunk 1]
```diff
@@ -107,6 +107,27 @@ class QPDFTokenizer
...
+        {
+            return this->type == tt_word;
+        }
+        bool
+        isWord(std::string const& value) const
+        {
+            return this->type == tt_word && this->value == value;
+        }
+        bool
+        isWord(std::string const& value1, std::string const& value2) const
```
Comment (jberkenbilt) [inline include/qpdf/QPDFTokenizer.hh:126]: I like the change except not this overload of isWord. You can just say `isWord("A") || isWord("B")` and not worry about the slight amount of extra work.

File: libqpdf/QPDF_linearization.cc [hunk 2]
```diff
@@ -121,13 +121,9 @@ QPDF::isLinearized()
         }
 
         QPDFTokenizer::Token t1 = readToken(this->m->file);
-        QPDFTokenizer::Token t2 = readToken(this->m->file);
```
Comment (jberkenbilt) [inline libqpdf/QPDF_linearization.cc:124]: This actually changes the meaning -- the old code unconditionally reads 4 tokens, while the new code may or may not read tokens depending on whether earlier tokens pass their checks. That means that this change changes where the file points to after the function is called. Have you determined whether this is safe?

Comment (jberkenbilt) [reply libqpdf/QPDF_linearization.cc:124]: There is a least one other example of that in this PR. Probably they are fine, but I haven't studied it.

**Review by jberkenbilt — CHANGES_REQUESTED** [review]

File: libqpdf/QPDF_linearization.cc (hunk 2 above)
Comment (m-holger) [reply libqpdf/QPDF_linearization.cc:124]: If the tokens don't pass the test either the loop is continued in which case another seek is called to explicitly set the file position before the next token is read, or the method returns false, in which case none of the callling code makes assumptions about the file position.

File: include/qpdf/QPDFTokenizer.hh (hunk 1 above)
Comment (m-holger) [reply include/qpdf/QPDFTokenizer.hh:126]: I'll take this out

**Review by jberkenbilt — APPROVED** [review]
Looks good. Thanks again for the contribution.

**Pull request merged**